projects/plugins/crm/includes/jpcrm-templating-placeholders.php

```diff
@@ -1755,6 +1755,27 @@ public function replace_placeholders(
 					
 					}
 
+					// If this is a Quote date key and is not set (Quote accepted or last viewed), let's print out a message saying the quote isn't accepted or viewed.
+					$possible_empty_quote_keys = array( 'quote-accepted', 'quote-accepted_date_str', 'quote-accepted_datetime_str', 'quote-lastviewed', 'quote-lastviewed_datetime_str', 'quote-lastviewed_date_str' );
+					if ( in_array( $key, $possible_empty_quote_keys, true ) ) {
+						// 82799 is 23hrs, 59mins, 59secs on Jan 1, 1970 (so a blank timestamp, allowing for systems to update the time within that day for any reason)
+						if ( empty( $replace_with ) || strtotime( $replace_with ) <= 82799 ) {
+							$replace_with = str_starts_with( $key, 'quote-accepted' ) ? __( 'Quote not yet accepted', 'zero-bs-crm' ) : __( 'Quote not yet viewed', 'zero-bs-crm' );
+						}
+					}
+
+					// We want to check if a quote field is a unix timestamp, so this light approach will check if the value is numeric initially to lower the searches,
+					// then check if its _datetime_str equivalent exists (unless it's the 'quote-date' key), confirming our current key is a timestamp.
+					// Then we can convert the replace_with value to a human readable format based on site date settings.
+					if ( str_starts_with( $key, 'quote-' ) && is_numeric( $replace_with ) ) {
```

Comment:
Wouldn't this basically assume any number (including floats) in a quote custom field is a date?

Reply:
The answer: yes, but you have a condition three lines down that filters those out.

Reply:
Yes that's right, my reasoning is that it is saving the `array_key_exists` check for fields that are more likely to be dates (well, numeric), instead of everything, or every quote field. I can easily remove it if it seems excessive.

Reply:
But, I'll be getting rid of this now given changes based on earlier comments.

Reply:
It's fine! I'll note that conditions fail fast, so if it was all in one `if` statement and the first condition failed, it wouldn't go on to the next, but nothing wrong with splitting it here.